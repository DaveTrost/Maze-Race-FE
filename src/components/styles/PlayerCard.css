@font-face {
  font-family: "joystix"; 
  src: url('../../assets/fonts/joystix.ttf')
}

.PlayerCard {
  display: flex;
  flex-direction: row;
  margin: 1vh 0vw;

  p {
    color: white;
    font-family: "joystix";
    font-size: 3vh;
    margin: 2vh 4vw;
  }

  div {
    border-radius: 1vw;
  }

  span {
    display: flex;
    flex-direction: row;
    width: 5vw;
    height: 5vh;
    align-items: center;
    justify-content: center;
    font-family: "Roboto-Bold";
    font-size: 3vh;
    -moz-box-shadow:    inset 0 0 5px 1px #000000;
    -webkit-box-shadow: inset 0 0 5px 1px #000000;
    box-shadow:         inset 0 0 5px 1px #000000;
  }
}