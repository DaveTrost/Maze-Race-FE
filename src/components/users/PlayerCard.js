import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/PlayerCard.css';


const PlayerCard = ({ name, color, symbol, ready }) => {
  console.log('TEST LOG****', ready);
  
  let finalReady = ready ? `${styles.PlayerReady} ${styles.PlayerReadyFinal}` : styles.PlayerReady;
  let finalCheck = ready ? `${styles.NotReady} ${styles.Ready}` : styles.NotReady;

  return (
    <div className={styles.PlayerCard}>
      <p>{name}</p>
      <div className={styles.PlayerSymbols}>
        <div className={finalReady} style={{ backgroundColor: color }}>
          <div>{symbol}</div>
        </div>
        <img className={finalCheck} src={ready ? 'src/assets/img/check.png' : 'src/assets/img/x.png'}></img>
      </div>
    </div>
  );

};

PlayerCard.propTypes = {
  name: PropTypes.string,
  color: PropTypes.string,
  symbol: PropTypes.string,
  ready: PropTypes.bool
};

export default PlayerCard;
