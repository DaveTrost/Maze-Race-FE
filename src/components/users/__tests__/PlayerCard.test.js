import React from 'react';
import { shallow } from 'enzyme';
import PlayerCard from '../PlayerCard.js';

describe('PlayerCard component', () => {
  it('renders PlayerCard', () => {
    const wrapper = shallow(<PlayerCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
