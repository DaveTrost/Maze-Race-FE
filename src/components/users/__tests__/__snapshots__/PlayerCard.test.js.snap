// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerCard component renders PlayerCard 1`] = `
<div
  className="PlayerCard"
>
  <p />
  <div
    className="PlayerSymbols"
  >
    <div
      className="PlayerReady"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <div />
    </div>
    <img
      className="NotReady"
      src="/path/image"
    />
  </div>
</div>
`;
